### ISSUE TITLE
Causalml ATE and Learning Rate Sensitivity

### ISSUE THREAD

**Issue by mravendi** [uber/causalml#247]
The default learning rate for LGBMRegressor in CausalML is 0.1. But I wanted to see how it would perform with other values
of learning rates.  As you can see below, it seems that the ATE can drastically change by changing the learning rate of the base learner. Anyone can help explain this behavior? 

Here is my setup. I generate some synthetic data:

```python
# train data
y, X, treatment, _, _, e = synthetic_data(mode=1, n=10000, p=5, sigma=1.0)
print(X.shape, y.shape, treatment.shape)

# test data
y_test, X_test, treatment_test, _, _, _ = synthetic_data(mode=1, n=5000, p=5, sigma=1.0)
print(X_test.shape, y_test.shape, treatment_test.shape)
```

and built a simple T-learner with different learning rates:

```python
for lr in [1e-5,1e-4,1e-3,1e-2,1e-1]:
    learner = LGBMRegressor(n_estimators=100, learning_rate=lr)
    learner_t = BaseTRegressor(learner=learner, control_name=0)
    ate_train, _, _ = learner_t.estimate_ate(X=X, treatment=treatment, y=y)
    
    # get ATE for test data
    te_test= learner_t.predict(X=X_test)
    ate_test = te_test.mean(axis=0)[0]
    
    print(f"learning rate: {lr}, train ATE: {ate_train[0]:.2f}")
    print(f"learning rate: {lr}, test ATE: {ate_test:.2f}")
        
    print("-"*50)
```

Here is the output:

```python
learning rate: 1e-05, train ATE: 0.87
learning rate: 1e-05, test ATE: 0.87
--------------------------------------------------
learning rate: 0.0001, train ATE: 0.87
learning rate: 0.0001, test ATE: 0.87
--------------------------------------------------
learning rate: 0.001, train ATE: 0.84
learning rate: 0.001, test ATE: 0.84
--------------------------------------------------
learning rate: 0.01, train ATE: 0.69
learning rate: 0.01, test ATE: 0.69
--------------------------------------------------
learning rate: 0.1, train ATE: 0.49
learning rate: 0.1, test ATE: 0.50
--------------------------------------------------
```

**Comment by ppstacy** [comment]
Hi @mravendi, this is interesting and thanks for sharing it with us.

When you use the different learning rates, have you looked at the model accuracy?  Usually, with a smaller learning_rate, it will need more num_iterations, which seems set at the same in the code you provided. I think it might be a more fair comparison to look at ATE when the model with similar accuracy. Let me know if this makes sense for you. Thanks!

**Comment by mravendi** [comment]
That is a good point. How do you measure the model accuracy here?

Regarding num_iteration, the default ```num_estimators =100``` in the CausalML. But I tired ```num_estimators =1000``` too. 

```python
for lr in [1e-5,1e-4,1e-3,1e-2,1e-1]:
    learner = LGBMRegressor(n_estimators=1000, learning_rate=lr)
    learner_t = BaseTRegressor(learner=learner, control_name=0)
    ate_train, _, _ = learner_t.estimate_ate(X=X, treatment=treatment, y=y)
    
    # get ATE for test data
    te_test= learner_t.predict(X=X_test)
    ate_test = te_test.mean(axis=0)[0]
    
    print(f"learning rate: {lr}, train ATE: {ate_train[0]:.2f}")
    print(f"learning rate: {lr}, test ATE: {ate_test:.2f}")
        
    print("-"*50)
```
Here is the output:

```python
learning rate: 1e-05, train ATE: 0.87
learning rate: 1e-05, test ATE: 0.87
--------------------------------------------------
learning rate: 0.0001, train ATE: 0.84
learning rate: 0.0001, test ATE: 0.84
--------------------------------------------------
learning rate: 0.001, train ATE: 0.69
learning rate: 0.001, test ATE: 0.69
--------------------------------------------------
learning rate: 0.01, train ATE: 0.49
learning rate: 0.01, test ATE: 0.50
--------------------------------------------------
learning rate: 0.1, train ATE: 0.47
learning rate: 0.1, test ATE: 0.48
--------------------------------------------------
```

Not much difference with before.

**Comment by ppstacy** [comment]
If you turn on the logging at INFO level it will return some basic evaluation metrics of the base learner. Below is the screenshot:

```
import logging
logger = logging.getLogger('Causalml')
logging.basicConfig(level=logging.INFO)
```

![image](https://user-images.githubusercontent.com/19823328/98875634-78959b00-2431-11eb-9a66-d9563ad764e8.png)

**Comment by mravendi** [comment]
Thanks, @ppstacy for the reply.

Clearly, there is a negative correlation between the accuracy metrics and the ATE values. It is strange since we expect the
worst-performing models (measured by metrics) to attain lower ATE values. 
So perhaps looking at the ATE as a measure of model performance is not enough and it should be in accordance with other metrics. 
 

```python
INFO:causalml:Error metrics for group 1
INFO:causalml:    RMSE   (Control):     1.0644
INFO:causalml:    RMSE (Treatment):     1.1303
INFO:causalml:   sMAPE   (Control):     0.8210
INFO:causalml:   sMAPE (Treatment):     0.5194
INFO:causalml:    Gini   (Control):     0.3902
INFO:causalml:    Gini (Treatment):     0.4476


learning rate: 1e-05, train ATE: 0.87
learning rate: 1e-05, test ATE: 0.87
--------------------------------------------------


INFO:causalml:Error metrics for group 1
INFO:causalml:    RMSE   (Control):     1.0629
INFO:causalml:    RMSE (Treatment):     1.1283
INFO:causalml:   sMAPE   (Control):     0.8204
INFO:causalml:   sMAPE (Treatment):     0.5188
INFO:causalml:    Gini   (Control):     0.3913
INFO:causalml:    Gini (Treatment):     0.4500


learning rate: 0.0001, train ATE: 0.87
learning rate: 0.0001, test ATE: 0.87
--------------------------------------------------


INFO:causalml:Error metrics for group 1
INFO:causalml:    RMSE   (Control):     1.0493
INFO:causalml:    RMSE (Treatment):     1.1097
INFO:causalml:   sMAPE   (Control):     0.8150
INFO:causalml:   sMAPE (Treatment):     0.5128
INFO:causalml:    Gini   (Control):     0.4318
INFO:causalml:    Gini (Treatment):     0.4618


learning rate: 0.001, train ATE: 0.84
learning rate: 0.001, test ATE: 0.84
--------------------------------------------------


INFO:causalml:Error metrics for group 1
INFO:causalml:    RMSE   (Control):     0.9729
INFO:causalml:    RMSE (Treatment):     1.0112
INFO:causalml:   sMAPE   (Control):     0.7887
INFO:causalml:   sMAPE (Treatment):     0.4824
INFO:causalml:    Gini   (Control):     0.4801
INFO:causalml:    Gini (Treatment):     0.5110


learning rate: 0.01, train ATE: 0.69
learning rate: 0.01, test ATE: 0.69
--------------------------------------------------


INFO:causalml:Error metrics for group 1
INFO:causalml:    RMSE   (Control):     0.7832
INFO:causalml:    RMSE (Treatment):     0.8115
INFO:causalml:   sMAPE   (Control):     0.7168
INFO:causalml:   sMAPE (Treatment):     0.4218
INFO:causalml:    Gini   (Control):     0.7212
INFO:causalml:    Gini (Treatment):     0.7237


learning rate: 0.1, train ATE: 0.49
learning rate: 0.1, test ATE: 0.50
--------------------------------------------------
```

**Comment by ppstacy** [comment]
First lower values of RMSE indicate better fit since it's a regression problem, so here isn't a negative correlation.
Second, I don't think it's a good idea only looking at ATE to measure model performance.  Maybe you can take a look of those example notebooks: [meta_learners_with_synthetic_data](https://github.com/uber/causalml/blob/master/examples/meta_learners_with_synthetic_data.ipynb) and [validation_with_tmle](https://github.com/uber/causalml/blob/master/examples/validation_with_tmle.ipynb).

**Comment by mravendi** [comment]
By negative correlation I mean lower values of RMSE have ** lowe ATE values.**
RMSE=1.x => ATE=0.87
RMSE=0.7x => ATE=0.49

But, intuitively, we expect when RMSE declines (better fit), ATE should increase.

**Comment by mravendi** [comment]
I am closing this issue as I think the point was made to not assume the ATE as the main criteria for model comparison.
